.circleci/db_schema_check

```diff
@@ -10,6 +10,34 @@ echo $GIT_STATUS
 if [[ $GIT_STATUS =~ "nothing to commit, working tree clean" ]]; then
   exit 0
 else
+  # This invocation ensures that our arrays split at new lines, and not BOTH new lines AND spaces.
+  # https://unix.stackexchange.com/questions/39473/command-substitution-splitting-on-newline-but-not-space
+  set -f
+  IFS='
+'
+
+  DIFF=`git diff db/schema.rb`
+  readarray -t DIFF_LINES <<<"$DIFF"
+  for line in "${DIFF_LINES[@]}" ; do
+    if [[ $line =~ ^\+ ]] ; then
+      if [[ $line =~ ^\+{3} ]] ; then
+        # +++ designates a file name, so we skip it.
+        echo ""
+      else
+        ADDITIONS+=($line)
+      fi
+    fi
+  done
+
+  # Reset the settings we edited above.
+  unset IFS
+  set +f
+
+  if [[ ${#ADDITIONS[@]} == 1 && ${ADDITIONS[0]} =~ ^\+ActiveRecord ]] ; then
+    echo "Rails version was updated, but there were no changes to any tables."
+    exit 0
+  fi
+
```

Comment:
I don't remember how to use an exit status after a command, but I think we could simplify this to something like the following:

```suggestion
  `git diff db/schema.rb | grep "^+[^+]" | grep -v "ActiveRecord::Schema\["`

  if [[ $? == 1 ]] ; then
    echo "Rails version was updated, but there were no changes to any tables."
    exit 0
  fi
```

Basically we leverage that addition lines start with `+` and then we filter out `ActiveRecord::Schema[`, `grep` will return 1 if there's no matches — meaning that no other changes were made in the file.

What happens if all that changes in a schema file is the `version:`? That's of course pretty rare, but do we end up failing the build in that case?

Reply:
Hey @kaspth, thank you for the review! This is definitely a lot simpler. I didn't know about `$?` returning a status, that's pretty useful. 

I tested this to check if a developer were to update their rails version AND and a model which they forget to migrate in the same go (an unlikely situation but possible nonetheless), and the check here covers that, so I'll try to push this change here soon.

Reply:
> I tested this to check if a developer were to update their rails version AND and a model which they forget to migrate in the same go (an unlikely situation but possible nonetheless), and the check here covers that, so I'll try to push this change here soon.

Ok, sounds good!